plugins/module_utils/k8s/service.py

```diff
@@ -435,19 +434,21 @@ def delete(
         resource: Resource,
         definition: Dict,
         existing: Optional[ResourceInstance] = None,
-    ) -> Dict:
+    ) -> Tuple[Dict, bool]:
```

Comment:
We can't change the return type of this function because it would break https://github.com/ansible-collections/kubernetes.core/blob/9f51fc0ef0319e84b54e8cbab28508307a11c34c/plugins/module_utils/k8s/runner.py#L98. The tests are likely passing because we aren't doing a deep analysis of the returned object. There should be no need for these changes, however (see comment below).